### PR TITLE
Translate hints

### LINKED ISSUE #3592: How to translate string with format
All the hints in the Registration Settings page are not translated.

See in OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml
https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml

`@T["— {0}", "Display a link to allow the users to register on the site."]`

How should it be generated in the .po files?
And does it have to be detected and generated by @lukaskabrt po-extractor tool?

### PR DESCRIPTION
**Pull request by agriffard**
Fixes #3592

Replace @T[] with a second argument as a string thta should be translated (mostly in hints with —).
Put `—` outside of the strings.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/Application/Edit.cshtml [hunk 1]
```diff
@@ -49,7 +49,7 @@
         <div class="custom-control custom-checkbox">
             <input type="checkbox" class="custom-control-input" asp-for="AllowImplicitFlow" data-toggle="collapse" data-target="#AllowImplicitFlowRecommendedHint" checked="@Model.AllowImplicitFlow">
             <label class="custom-control-label" asp-for="AllowImplicitFlow">@T["Allow Implicit Flow"]</label>
-            <span class="hint">@T["— {0}", "More info:"] <a href="http://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowAuth">http://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowAuth</a></span>
+            <span class="hint">— @T["More info:"] <a href="http://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowAuth">http://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowAuth</a></span>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/Application/Edit.cshtml]: @hishamco how will this kind of thing work when localized in RTL ? will the link and text and dash work correctly?

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml [hunk 2]
```diff
@@ -99,15 +99,15 @@
...
+        <span class="hint"><a href="http://openid.net/specs/openid-connect-core-1_0.html#id_tokenExample" target="_blank">— @T["example:"] http://openid.net/specs/openid-connect-core-1_0.html#id_tokenExample</a></span>
     </div>
 </fieldset>
 
 <fieldset class="form-group" asp-validation-class-for="UseIdTokenTokenFlow">
     <div class="custom-control custom-checkbox">
         <input type="checkbox" class="custom-control-input" asp-for="UseIdTokenTokenFlow">
         <label class="custom-control-label" asp-for="UseIdTokenTokenFlow">@T["Use 'id_token token' response type"]</label>
-        <span class="hint"><a href="http://openid.net/specs/openid-connect-core-1_0.html#id_token-tokenExample" target="_blank">@T["— {0}", "example:"] http://openid.net/specs/openid-connect-core-1_0.html#id_token-tokenExample</a></span>
+        <span class="hint"><a href="http://openid.net/specs/openid-connect-core-1_0.html#id_token-tokenExample" target="_blank">— @T["example:"] http://openid.net/specs/openid-connect-core-1_0.html#id_token-tokenExample</a></span>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: @hishamco And this one?

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/Application/Edit.cshtml (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/Application/Edit.cshtml]: It should work without any issue, coz the page direction will control the flow

> will the link and text and dash work correctly?

Yes, if all the text after and before are translated

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml (hunk 2 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: Should work too

**Pull request merged**